### PR TITLE
Refactor mock server

### PR DESCRIPTION
**Pull request by jelly**
Follow up idea, get rid of the hardcoded ports, just pick a random unused port by binding on `0` and use that.

Personally dislike the location of `test_mock_server` but I am not sure yet how I feel about having `test/__init__.py` and it's consequences.

### CODE REVIEW

File: task/test-github [hunk 1]
```diff
@@ -19,92 +19,59 @@
...
-                    self.replyData("", status=304)
-                else:
-                    self.replyJson({"user": "blah"}, headers={"Last-Modified": "Thu, 05 Jul 2012 15:31:30 GMT"})
-            elif parsed.path == "/orgs/cockpit-project/teams/contributors/members":
-                self.replyJson([
-                    {"login": "one"},
-                    {"login": "two"},
-                    {"login": "three"}
-                ])
+DATA = [{"number": "5", "state": "open", "created_at": "2011-04-22T13:33:48Z"},
```
Comment (martinpitt) [inline task/test-github:32]: Perhaps GITHUB_DATA or GITHUB_ISSUES or so?

File: task/test_mock_server.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import http.server
+import json
+import multiprocessing
+
+
+class MockServer(http.server.HTTPServer):
+    def __init__(self, address, handler, data=None):
+        self.address = address
+        self.handler = handler
+        self.data = data
```
[the remark this answers is not in the document]
Comment (jelly) [reply task/test_mock_server.py:29]: It is used, this is how the handler accesses and mutates the data.

[the remark this answers is not in the document]
Comment (martinpitt) [reply task/test_mock_server.py:29]: Hmm, can you please point me to it? I don't see any `self.data` other than this one.

[the remark this answers is not in the document]
Comment (jelly) [reply task/test_mock_server.py:29]: Via via here https://github.com/cockpit-project/bots/pull/5187/files#diff-d2005d22123d681b75e10a1d27f5eac2f03df14fdda55809c4d4f1f282a99672R53

**Review by martinpitt — APPROVED** [review]
Ack, thanks!

**Pull request merged**